config/babel-register/babel-hooks.mjs

```diff
@@ -71,7 +71,16 @@ async function transpile(source, context) {
  * the Node.js default resolve hook after the last user-supplied resolve hook
  */
 export async function resolve(specifier, context, nextResolve) {
-    return nextResolve(specifier, context);
+    try {
+        return await nextResolve(specifier, context);
+    } catch (err) {
+        if (err.code === 'ERR_MODULE_NOT_FOUND') {
```

Comment:
Can you comment why you need this specific case?

Reply:
We first try to resolve the path of the imported module as a `.js` file. However, if such module is a `.ts` file, we retry the resolver substituing the extension from `.js` to `.ts`.

Reply:
that's what I inferred, I think it's worth adding your response as a comment in the code